### PR TITLE
chore(nim-chronos): bump submodule

### LINKED ISSUE #2849: bug: cannot compile with gcc-14 `-Wincompatible-pointer-types`
### Problem

Cannot compile with gcc-14

See similar

- https://github.com/status-im/status-desktop/issues/15300
- https://github.com/status-im/status-desktop/pull/15250

It looks like we can just ignore it.

### Screenshots/logs

```
/home/fryorcraken/src/waku-org/nwaku/vendor/nim-serialization/serialization/object_serialization.nim(154, 32) template/generic instantiation of `enumAllSerializedFieldsImpl` from here
/home/fryorcraken/src/waku-org/nwaku/vendor/nim-toml-serialization/toml_serialization/private/array_reader.nim(53, 21) template/generic instantiation of `readValue` from here
/home/fryorcraken/src/waku-org/nwaku/vendor/nim-toml-serialization/toml_serialization/reader.nim(224, 14) template/generic instantiation of `readValue` from here
/home/fryorcraken/src/waku-org/nwaku/vendor/nim-toml-serialization/toml_serialization/reader.nim(217, 10) Warning: setLen can potentially expand the sequence, but the element type 'ProtectedTopic' doesn't have a valid default value [UnsafeSetLen]
/home/fryorcraken/src/waku-org/nwaku/vendor/nim-chronos/chronos/streams/tlsstream.nim: In function ‘newTLSClientAsyncStream__OOZOOZvendorZnim45chronosZchronosZappsZhttpZhttpclient_995’:
/home/fryorcraken/src/waku-org/nwaku/vendor/nim-chronos/chronos/streams/tlsstream.nim:514:100: error: passing argument 2 of ‘sslEngineSetX509__OOZOOZvendorZnim45bearsslZbearsslZabiZbearssl95ssl_475’ from incompatible pointer type [-Wincompatible-pointer-types]
  514 |     sslEngineSetX509(res.ccontext.eng, addr res.xwc.vtable)
      |                                                                                                    ^                 
      |                                                                                                    |
      |                                                                                                    const br_x509_class ** {aka const struct br_x509_class_ **}
nimcache/release/wakunode2/@m..@s..@svendor@snim-chronos@schronos@sstreams@stlsstream.nim.c:716:148: note: expected ‘br_x509_class **’ {aka ‘struct br_x509_class_ **’} but argument is of type ‘const br_x509_class **’ {aka ‘const struct br_x509_class_ **’}
  716 | 9__OOZOOZvendorZnim45bearsslZbearsslZabiZbearssl95ssl_475)(br_ssl_engine_context* cc, br_x509_class** x509ctx_0);
      |                                                                                       ~~~~~~~~~~~~~~~~^~~~~~~~~

Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3 -pthread -I"/home/fryorcraken/src/waku-org/nwaku/vendor/nim-libbacktrace" -I"/home/fryorcraken/src/waku-org/nwaku/vendor/nim-libbacktrace/install/usr/include" -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-bearssl/bearssl/abi -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-bearssl/bearssl/abi/../csources/src/ -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-bearssl/bearssl/abi/../csources/inc/ -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-bearssl/bearssl/abi/../csources/tools/ -DBR_USE_UNIX_TIME=1 -DBR_USE_URANDOM=1 -DBR_LE_UNALIGNED=1 -DBR_64=1  -DBR_amd64=1 -DBR_INT128=1 -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-zlib/zlib/csources -DHAVE_UNISTD_H -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-nat-traversal/vendor/miniupnp/miniupnpc/include -I/home/fryorcraken/src/waku-org/nwaku/vendor/nim-nat-traversal/vendor/libnatpmp-upstream -DENABLE_STRNATPMPERR -march=native -g3 -Og -O3 -fno-strict-aliasing -fno-ident   -I/home/fryorcraken/src/waku-org/nwaku/vendor/nimbus-build-system/vendor/Nim/lib -I/home/fryorcraken/src/waku-org/nwaku/apps/wakunode2 -o nimcache/release/wakunode2/@m..@s..@svendor@snim-chronos@schronos@sstreams@stlsstream.nim.c.o nimcache/release/wakunode2/@m..@s..@svendor@snim-chronos@schronos@sstreams@stlsstream.nim.c' failed with exit code: 1


stack trace: (most recent call last)
/home/fryorcraken/src/waku-org/nwaku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(429, 18)
/home/fryorcraken/src/waku-org/nwaku/waku.nimble(77, 15) wakunode2Task
/home/fryorcraken/src/waku-org/nwaku/waku.nimble(35, 8) buildBinary
/home/fryorcraken/src/waku-org/nwaku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(273, 7) exec
/home/fryorcraken/src/waku-org/nwaku/vendor/nimbus-build-system/vendor/Nim/lib/system/nimscript.nim(273, 7) Error: unhandled exception: FAILED: nim c --out:build/wakunode2  --verbosity:0 --hints:off -d:chronicles_log_level=TRACE -d:git_version="v0.29.0-15-g010501" -d:release --passL:librln
```

### nwaku version/commit hash
`01050138c66206e48d9d5472e65b6868058a57d0`

### PR DESCRIPTION
**Pull request by fryorcraken**
# Description

Bump nim-chronos submodule to remove the incompatible-pointer-types error introduced with [gcc-14](https://gcc.gnu.org/gcc-14/porting_to.html).

Interesting commit is 1b9d9253e89445d585d0fff39cc0d19254fdfd0d

# Changes

- [x] bump nim-chronos submodule

## How to test

1. Build and run

## Issue

Fixes #2849

### CODE REVIEW

File: config.nims [hunk 1]
```diff
@@ -105,3 +105,7 @@ if defined(android):
   switch("passC", "--sysroot=" & sysRoot)
   switch("passL", "--sysroot=" & sysRoot)
   switch("cincludes", sysRoot & "/usr/include/")
+
+when defined(gcc):
+  # GCC 14+ introduces new strictness for pointer types that not all nim libraries are compatible with
```
Comment (Ivansete-status) [inline config.nims:110]: I wonder if we could indicate the libraries that has this incompatibility issue

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it! :100: 
I just added a nitpick comment in case we have that info in hand.
Cheers

**Comment by gabrielmer** [comment]
I see in the issue that the compilation error comes from `nim-chronos`. And I see that they fixed it 3 weeks ago https://github.com/status-im/nim-chronos/commit/1b9d9253e89445d585d0fff39cc0d19254fdfd0d

Isn't it better to just upgrade `nim-chronos` to a newer version that includes this fix?

**Comment by fryorcraken** [comment]
> I see in the issue that the compilation error comes from `nim-chronos`. And I see that they fixed it 3 weeks ago [status-im/nim-chronos@1b9d925](https://github.com/status-im/nim-chronos/commit/1b9d9253e89445d585d0fff39cc0d19254fdfd0d)
> 
> Isn't it better to just upgrade `nim-chronos` to a newer version that includes this fix?

Indeed, there was much more submodules in status-desktop that were failing but only nim-chronos for nwaku. now bumped and fixed.

**Review by gabrielmer — APPROVED** [review]
Thanks so much!

**Review by NagyZoltanPeter — APPROVED** [review]
Yeah, cool! 
Also fixes stringify of Duration type. I think it fixes printing 0ns which printed empty string before.

**Pull request merged**